### PR TITLE
Fixes banned reagents being transferable

### PR DESCRIPTION
**Pull request by Sometinyprick**
Does what the title says
Meant for spray bottles
Thanks to RemieRichards for showing me how to do this a better way (and basically doing the hard part for me)
and  to oranges for showing me how to make my prs less full of shit commits.

### CODE REVIEW

**Comment by Razharas** [comment]
I actually thought its a proc inside the object that is refusing the reagent, not in the one that transfers
Meh

**Comment by Fox-McCloud** [comment]
I believe this needs to be done for syringes, as well.

**Comment by Sometinyprick** [comment]
It has been done for syringes?
Look at the files, I added a check for syringes.

**Comment by Fox-McCloud** [comment]
Derp; I'm an idiot. Sorry!

**Comment by Fox-McCloud** [comment]
This doesn't work.

What happens: if you try to transfer Flourosulfuric acid (with or without anything else with it) it fails and gives the message.

Try to transfer ANY other reagent, from water to lexorin.....(even regular sufluric) and it won't transfer at all; no message is generated with sulfuric.

**Comment by Sometinyprick** [comment]
It worked fine when I tested it.
I will test it again when I get a chance.
Thanks for pointing this out though.

**Comment by Fox-McCloud** [comment]
@Sometinyprick 

I used a beaker; I believe I got similar results with a syringe (but don't quote me on that).

**Comment by Sometinyprick** [comment]
Yea nothing is transferring to spray bottles, good thing you noticed this. I will try and fix it when I get home

File: code/modules/reagents/reagent_containers.dm [hunk 1]
```diff
@@ -77,3 +77,11 @@
 
 			return 0
 	return 1
+obj/item/weapon/reagent_containers/proc/reject_bad_chem(var/obj/item/weapon/reagent_containers/RC,mob/user)
+    if(istype(RC))
+        for(var/bad_reg in RC.banned_reagents)
+            if(reagents.has_reagent(bad_reg, 1))
+                user << "<span class='warning'> A chemical in [src] is far too dangerous to transfer to [RC]!</span>"
+            return 1
```
Comment (RemieRichards) [inline code/modules/reagents/reagent_containers.dm]: Move this under the if. this is causing it to fail to transfer any reagents.

Comment (Sometinyprick) [inline code/modules/reagents/reagent_containers.dm]: Oh, that was dumb of me.
Thanks!

**Comment by Fox-McCloud** [comment]
Still doesn't work.

**Comment by Sometinyprick** [comment]
Everything is working fine on my end.
What issues are you running into?

**Comment by Sometinyprick** [comment]
Eh I did some more testing for this and it still allows you to do some stupid stuff (Like mixing one unit of water and facid then adding it to the spray will bypass it) and there is a bunch of other things you can do to transfer it anyway.
I'm going to try do it a different way, 
Hopefully I don't fuck that up too.

**Comment by Delimusca** [comment]
i think the ideal way to handle acid is to give it a minimum volume that also affects melt chance.
obviously smoke will still be awful but i'm working on making a hybrid of goonsmoke and ours.

**Comment by Fox-McCloud** [comment]
@Delimusca 

I'd just go with full on Goonsmoke if you're going to do that; the single application means that it resolves the issue of multiproccing and such--either case, this isn't the PR for it; would love to discuss this with you further, though.

**Pull request closed without merging**